krun.py

```diff
@@ -380,26 +388,45 @@ def sanity_check_user_change(platform):
         vd.run_exec(ep, bench_name, iterations, param, SANITY_CHECK_HEAP_KB)
 
     try:
-        times = check_and_parse_execution_results(stdout, stderr, rc)
-    except ExecutionFailed as e:
-        fatal("%s sanity check failed: %s" % (bench_name, e.message))
-
+        times = util.check_and_parse_execution_results(stdout, stderr, rc)
+    except util.ExecutionFailed as e:
+        util.fatal("%s sanity check failed: %s" % (bench_name, e.message))
+
+
+def create_arg_parser():
+    """Create a parser to process command-line options.
+    """
+    parser = argparse.ArgumentParser(description='Benchmark, running many fresh processes.')
+    parser.add_argument('--resume', '-r', action='store_true', default=False,
+                        dest='resume', required=False,
+                        help='Resume benchmarking if interrupted')
+    parser.add_argument('--reboot', '-b', action='store_true', default=False,
+                        dest='reboot', required=False,
+                        help='Reboot after every execution')
+    parser.add_argument('config', action="store", # Required by default.
+                        metavar='FILENAME',
+                        help='krun configuration file, e.g. experiment.krun')
+    return parser
 
 def main():
-    try:
-        config_file = sys.argv[1]
-    except IndexError:
-        usage()
+    parser = create_arg_parser()
+    args = parser.parse_args()
+
+    if not args.config.endswith(".krun"):
+        usage(parser)
 
-    if not config_file.endswith(".krun"):
-        usage()
+    try:
+        if os.stat(args.config).st_size <= 0:
```

Comment:
Do we need this check. I think it would bomb out later when it finds config variables missing.